Gems/Atom/Tools/AtomToolsFramework/Code/Source/DynamicNode/DynamicNode.cpp

```diff
@@ -82,5 +94,20 @@ namespace AtomToolsFramework
                 }
             }
         }
+
+        for (const auto& slotConfig : m_config.m_propertySlots)
+        {
+            for (const auto& supportedDataType : slotConfig.m_supportedDataTypes)
+            {
+                if (auto dataType = GetGraphContext()->GetDataType(supportedDataType))
+                {
+                    const AZStd::any& defaultValue =
+                        !slotConfig.m_defaultValue.empty() ? slotConfig.m_defaultValue : dataType->GetDefaultValue();
+                    RegisterSlot(GraphModel::SlotDefinition::CreateProperty(
+                        slotConfig.m_name, slotConfig.m_displayName, dataType, defaultValue, slotConfig.m_description));
+                    break;
+                }
+            }
+        }
```

Comment:
The same comment applies for m_outputSlots above as well.

Reply:
For output and property slots, only one data type gets registered. It's breaking immediately after the first type gets registered. Input slots should support multiple types, different scalar and vector types that can be assigned to each other. It might be better to create a primary data type or default data type for the slot configuration to be more explicit. These loops are treating the first compatible type as the primary type for the slot.